### ISSUE TITLE
Crash when creating a health food store

### ISSUE THREAD

**Issue by msearle5** [NullCGT/SpliceHack#28]
Health food stores contain the object type VEGETARIAN_CLASS, which is special cased by mkshobj_at - it should never call mkobj_at with that type, and if it does it will fall off the end of the array and crash. However, there was a missing else causing it to continue past the mkveggy_at() and call mkobj_at as well.
This can be fixed with the following patch:

517c517
<         if (Is_blackmarket(&u.uz)) {
---
>         else if (Is_blackmarket(&u.uz)) {

**Comment by msearle5** [comment]
517c517
<         if (Is_blackmarket(&u.uz)) {
---
>         else if (Is_blackmarket(&u.uz)) {

OK, that got mangled. Trying it again.

**Comment by msearle5** [comment]
``517c517
<         if (Is_blackmarket(&u.uz)) {
---
>         else if (Is_blackmarket(&u.uz)) {

**Comment by msearle5** [comment]
```
517c517
<         if (Is_blackmarket(&u.uz)) {
---
>         else if (Is_blackmarket(&u.uz)) {
```
OK, like that :)

**Comment by NullCGT** [comment]
Oh my goodness, thank you so much! This explains the segfault someone got today on the second level of the  dungeon in mkobj, I've been pulling my hair out trying to find out what could have possibly caused it because I never touched any mkobj code or altered object classes. I would have never thought to look in there!

Fixed by commit 01f40dcbfe6d83b80ae8d3d889349532453a916c.
